packages/client/src/Client.ts

```diff
@@ -40,14 +49,19 @@ export type CheckVersionBound<Q> = Q extends VersionBound<infer V> ? (
         [ErrorMessage]:
           `Your SDK requires a semver compatible version with ${V}. You have ${MaxOsdkVersion}. Update your package.json`;
       }
-    // Q
   )
   : Q;
 
 export interface Client extends SharedClient<MinimalClient> {
   <Q extends MinObjectDef<any, any>>(
     o: Q,
-  ): CompileTimeMetadata<Q>["objectSet"];
+  ): unknown extends CompileTimeMetadata<Q>["objectSet"] ? ObjectSet<Q>
```

Comment:
Why did we need this change?

Reply:
I think this is similar to the issue you saw recently where the type complexity ends up failing us. If I recall it was no longer able to determine if the primaryKey was the same type

Reply:
Ah yea that was the same issue I ran into